### ISSUE TITLE
Fix S3240: Use the '?:' operator here

### ISSUE THREAD

**Issue by herecydev** [SonarSource/sonar-csharp#1467]
### Description

Ternary suggestion should not be presented if the ternary paths return void.

### Repro steps

```c#
if(number.TryTake(out var number))
DoSomethingVoid();
else
DoSomeMoreVoids();
```

### Expected behavior

As this cannot be simplified (at least to my understanding of C#) because the 2 paths return void, then no violation should be raised

### Actual behavior

A violation is raised that has no fix.

**Comment by Evangelink** [comment]
Hi @herecydev,

We don't have such rule (nor S3420 nor a rule suggesting to use the ternary operator). Could you please double check your code to be sure about the issue reported?

Cheers,
Amaury

**Comment by valhristov** [comment]
@Evangelink I updated the rule title, it is about S3240...

**Comment by Evangelink** [comment]
Jira search... I was looking with ternary operator and the rule didn't show up...

**Comment by herecydev** [comment]
Apologies for mistyping the rule. I'll check tomorrow what version's we're using and report back

**Comment by Evangelink** [comment]
Hi @herecydev,

Sorry for this late ping but could you let me know if you are still experiencing this issue?

**Comment by Evangelink** [comment]
Closing the issue. Please feel free to re-open or re-create if you still experience the issue.
